packages/gafl-webapp-service/src/handlers/__tests__/page-handler.spec.js

```diff
@@ -145,51 +145,27 @@ describe('The page handler function', () => {
     )
   })
 
-  it.each([['/go/somewhere'], ['/go/somewhere/else']])(
-    'GetDataRedirect being thrown will pass url to be decorated to addLanguageCodeToUri',
-    async redirectUri => {
-      journeyDefinition.push({ current: { page: 'view' }, backLink: '/previous/page' })
-      const getData = () => {
-        throw new GetDataRedirect(redirectUri)
-      }
-      const { get } = pageHandler(null, 'view', '/next/page', getData)
-      const toolkit = getMockToolkit()
-      const request = getMockRequest()
-
-      await get(request, toolkit)
-
-      expect(addLanguageCodeToUri).toHaveBeenCalledWith(request, redirectUri)
-    }
-  )
-
-  it('GetDataRedirect being thrown will use addLanguageCodeToUri to decorate the redirect target', async () => {
+  it('GetDataRedirect being thrown will redirectWithLanguageCode to the target', async () => {
     journeyDefinition.push({ current: { page: 'view' }, backLink: '/previous/page' })
+    const url = Symbol('/go/somewhere/else')
     const getData = () => {
-      throw new GetDataRedirect('/go/somewhere/else')
+      throw new GetDataRedirect(url)
     }
     const { get } = pageHandler(null, 'view', '/next/page', getData)
-    const returnValue = Symbol('/previous/page')
-    addLanguageCodeToUri.mockReturnValueOnce(returnValue)
+    const request = getMockRequest()
     const toolkit = getMockToolkit()
 
-    await get(getMockRequest(), toolkit)
-
-    expect(toolkit.redirect).toHaveBeenCalledWith(returnValue)
-  })
+    await get(request, toolkit)
 
-  it('error calls addLanguageCodeToUri with request', async () => {
-    const { error } = pageHandler('', 'view')
-    const request = getMockRequest()
-    await error(request, getMockToolkit(), { details: [] })
-    expect(addLanguageCodeToUri).toHaveBeenCalledWith(request)
+    expect(toolkit.redirectWithLanguageCode).toHaveBeenCalledWith(url)
   })
 
-  it.each([['/route/one'], ['/route/sixty-six']])('error redirects to uri decorated by addLanguageCodeToUri', async url => {
+  it.each([['/route/one'], ['/route/sixty-six']])('error redirects with language code', async url => {
     const { error } = pageHandler('', 'view')
+    const request = getMockRequest(undefined, url)
     const toolkit = getMockToolkit()
-    addLanguageCodeToUri.mockReturnValueOnce(`Redirect to url ${url}`)
-    await error(getMockRequest(undefined, url), toolkit, { details: [] })
-    expect(toolkit.redirect).toHaveBeenCalledWith(`Redirect to url ${url}`)
+    await error(request, toolkit, { details: [] })
+    expect(toolkit.redirectWithLanguageCode).toHaveBeenCalledWith()
```

Comment:
`toHaveBeenCalled` would be semantically accurate here. if you want to check it was called with no arguments (which might be a good test) then I'd suggest examining the mock method that there were no arguments passed.

Reply:
Yeah, I flipflopped between `toHaveBeenCalled()` and `toHaveBeenCalledWith()` with no arguments. I'll have a look at examining the method.

Reply:
On further testing it looks like `toHaveBeenCalledWith()` does check if `redirectWithLanguageCode` is called with no arguments, and fails if it is called with arguments. Personally I am happy with this and feel like it is more consistent with the rest of the tests, but can rewrite if you would rather we test it in a different way.